src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -284,9 +284,13 @@ public override bool IsReadOnly
                     {
                         _isReadOnly = BindingMetadata.IsReadOnly;
                     }
+                    else if (MetadataKind == ModelMetadataKind.Property)
+                    {
+                        _isReadOnly = _details.PropertySetter == null;
+                    }
```

Comment:
This block should probably come first (ignore `BindingMetadata.IsReadOnly` if it's Type metadata).